hypervisor/arch/x86/configs/vacpi.c

```diff
@@ -108,7 +130,19 @@ void build_vacpi(struct acrn_vm *vm)
 	struct acpi_table_mcfg *mcfg;
 	struct acpi_table_madt *madt;
 	struct acpi_madt_local_apic *lapic;
-	uint16_t i;
+	uint16_t i, table_entry = 0U;
+	bool pt_tpm2_acpitable = false;
+
+#if PT_MMIODEV_PRELAUNCHED_VM
+	struct acrn_vm_config *vm_config = get_vm_config(vm->vm_id);
+	for (i = 0U; i < MAX_MMIO_DEV_NUM; i++) {
+		if ((vm_config->mmiodevs[i].size != 0UL) &&
+				(vm_config->mmiodevs[i].size == TPM_BUFFER_SIZE)) {
```

Comment:
WHy we have this kind of condition check?

Reply:
Loop for each mmio devices, and do necessary operations (like vACPI) for each of the mmio device.  This check is completely wrong,

Reply:
Now there's no sanitize check about this config. We could remove this check when we have this check.
How could we know what vACPI table should fill before we know the type of the MMIO device ?
The ACPI Table doesn't have the same format for each MMIO device.

Reply:
The configuration file can identify the ACPI method that we want to PT. This is so low level bug.

Reply:
The configuration file could not identify the ACPI method that we want to PT for a specified VM.